src/hotspot/share/services/nmtCommon.hpp

```diff
@@ -59,25 +55,16 @@
 //             - malloc headers are used
 //             - malloc call site table is allocated and used
 //
-// Valid state transitions:
-//
-// unknown ----> off
-//          |
-//          |--> summary --
-//          |              |
-//          |--> detail  --+--> minimal
-//
 
 
 // Please keep relation of numerical values!
-// unknown < off < minimal < summary < detail
+// unknown < off < summary < detail
 //
 enum NMT_TrackingLevel {
-  NMT_unknown = 0,
-  NMT_off     = 1,
-  NMT_minimal = 2,
-  NMT_summary = 3,
-  NMT_detail  = 4
+  NMT_unknown,
+  NMT_off,
+  NMT_summary,
+  NMT_detail
```

Comment:
I'm fine with removing the explicit numbers here, but could you please add
```
STATIC_ASSERT(off > unknown)
STATIC_ASSERT(summary > off)
STATIC_ASSERT(detail > summary)
```
somewhere? (Maybe in the cpp file, its not such an eyesore there).

Reply:
Good idea, fixed.